### PR TITLE
docs: Add Jest polyfill workaround to troubleshooting docs

### PR DESCRIPTION
**Pull request by slaymance**
#### Description of changes
This PR adds troubleshooting steps when users of the Jest testing framework run into the following error when using the `@aws-amplify/ui-react` package:
```
window.URL.createObjectURL is not a function
```

#### Issue #, if available
#1647 

#### Description of how you validated changes
You may validate this polyfill works by deleting our own instance of this workaround in `packages/react/jest.setup.ts` and attempting to run our Jest tests with `yarn react test`.

#### Checklist

- [x] PR description included
- [x] `yarn test` passes
- [x] Tests are updated - N/A
- [x] Relevant documentation is changed or added (and PR referenced)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: docs/src/pages/getting-started/installation/troubleshooting.react.mdx [hunk 1]
```diff
@@ -90,3 +90,26 @@ export default defineConfig({
     "skipLibCheck": true,
 ...
 ```
+
+### Jest
+
+In latest versions of `@aws-amplify/ui-react`, users of the Jest testing framework may run into the following error
```
Comment (calebpollman) [inline docs/src/pages/getting-started/installation/troubleshooting.react.mdx:96]: Should we state the specific version this started in and up? e.g. `in versions of `@aws-amplify/ui-react` 2.15.3 and above` (not sure if that is the correct version we introduced `MapView` in btw 😉)

Comment (slaymance) [reply docs/src/pages/getting-started/installation/troubleshooting.react.mdx:96]: Yes that's much better.

**Review by calebpollman — APPROVED** [review]

File: docs/src/pages/getting-started/installation/troubleshooting.react.mdx [hunk 2]
```diff
@@ -90,3 +90,26 @@ export default defineConfig({
...
+**1.** Navigate to or create a [Jest setup file](https://jestjs.io/docs/configuration#setupfilesafterenv-array) for your project.
+**2.** Add the following code to polyfill the unrecognized function in your Jest setup file:
+
+```js
+if (typeof window.URL.createObjectURL === 'undefined') {
+  window.URL.createObjectURL = jest.fn();
+}
+```
+
+This is a known problem when using the `jsdom` library (a dependency of Jest) with a package that uses the unrecognized
```
Comment (reesscot) [inline docs/src/pages/getting-started/installation/troubleshooting.react.mdx:114]:
```suggestion
This is a known problem when using the `jsdom` library (a dependency of Jest) with a package that uses an unrecognized
```

**Review by reesscot — APPROVED** [review]
LGTM, one minor wording nit

File: docs/src/pages/getting-started/installation/troubleshooting.react.mdx [hunk 3]
```diff
@@ -90,3 +90,26 @@ export default defineConfig({
     "skipLibCheck": true,
 ...
 ```
+
+### Jest
+
+As of v2.15.0 of `@aws-amplify/ui-react` which included the release of Geo components, users of the Jest testing
+framework may run into the following error when attempting to run tests:
```
Comment (wlee221) [inline docs/src/pages/getting-started/installation/troubleshooting.react.mdx:97]: Very minor nit: this newline should be deleted and let markdown handle newlines natively. Same with line 114.

Fyi, vs code has "Toggle word wrap" options you can use for local dev helps.

File: .changeset/angry-feet-cross.md [hunk 4]
```diff
@@ -0,0 +1,5 @@
+---
+'docs': patch
+---
+
+Add React troubleshooting steps for Jest polyfill error
```
Comment (wlee221) [inline .changeset/angry-feet-cross.md:5]: This changeset should be removed 🙏

**Review by wlee221 — APPROVED** [review]

**Pull request merged**